tests/Doctrine/Tests/Common/DataFixtures/Purger/ORMPurgerForeignKeyCheckTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Common\DataFixtures;
+
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\Common\EventManager;
+use Doctrine\DBAL\Configuration;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\ORMSetup;
+use Doctrine\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\Tests\Common\DataFixtures\TestEntity\Link;
+
+class ORMPurgerForeignKeyCheckTest extends BaseTest
+{
+    public const TEST_CLASS_NAME = Link::class;
+    public const TEST_TABLE_NAME = 'link';
+
+    /** @return MappingDriver */
```

Comment:
Why not use a native return type hint here?

Reply:
I was doing it like the BaseTest::getMock*() functions. No return type, but hinting the real class.

I guess like ORMExecutorTest::getMockFixture() is the way to go?

Reply:
Yes, because `BaseTest::getMock*()` do not return _phpunit_ mocks, unlike what you are doing.